sdk/template/azure-template-stress/README.md

```diff
@@ -0,0 +1,228 @@
+You may additionally consider assigning different `APPLICATIONINSIGHTS_ROLE_NAME`.
+In any case, you may need to adjust the workbook to accomodate those changes and break down metrics by role instance.
+
+In addition to `test.run_duration`, we're also collecting:
+- [reactor schedulers metrics](https://github.com/reactor/reactor-core/blob/main/docs/asciidoc/metrics.adoc)
+- JVM metrics measured by ApplicationInsights agent:
+    - Normalized CPU percentage (based on [OperatingSystemMXBean](https://docs.oracle.com/javase/8/docs/api/java/lang/management/OperatingSystemMXBean.html) and [RuntimeMXBean](https://docs.oracle.com/javase/8/docs/api/java/lang/management/RuntimeMXBean.html)))
+      Note: if 0.5 CPU is configured for pod, maximum normalized CPU percentage for it would be 50%.
+    - GC time (based on [GarbageCollectorMXBean](https://docs.oracle.com/javase/8/docs/api/java/lang/management/GarbageCollectorMXBean.html))
+    - Heap memory usage (based on [MemoryMXBean](https://docs.oracle.com/javase/8/docs/api/java/lang/management/MemoryMXBean.html))
+    - Thread count (based on [ThreadMXBean](https://docs.oracle.com/javase/8/docs/api/java/lang/management/ThreadMXBean.html))
+
+### Stress test workbook
+
+[Stress test workbook](https://ms.portal.azure.com/#@microsoft.onmicrosoft.com/resource/subscriptions/faa080af-c1d8-40ad-9cce-e1a450ca5b57/resourceGroups/rg-stress-cluster-pg/providers/Microsoft.Insights/components/stress-pg-ai-s7b6dif73rup6/workbooks)
+shows a summary of a test run.
+
+First, select a time range and run from the list, then check the report:
+- `Test summary` contains key test parameters and key counters (total number of operations, errors, etc.)
+- Tst operation success rate, latency and error rate
+- CPU and memory utilization, number of threads and time spent in GC
+- Warnings, errors, and exceptions in logs. Note logs and traces are sampled (at 1%) rate, so you won't see every error there
+
+Since you're changing the chart name, you would need to update the workbook to use `java-your-service-name` instead of `java-template`.
+Then you'd need to create a new workbook for your service, follow
+[Azure Monitor workbook documentation](https://learn.microsoft.com/azure/azure-monitor/visualize/workbooks-create-workbook) for more details.
+Then you can import json file from `workbooks` folder.
+
+## Writing useful tests
+
+Stress tests are intended to detect reliability and resiliency issues:
+- bugs in retry policy
+- graceful degradation under high load and transient failures
+- memory leaks, thread pool starvation, etc
+
+While we don't have generic solution for fault injection, check out [Chaos mesh](https://github.com/Azure/azure-sdk-tools/blob/main/tools/stress-cluster/chaos/README.md#chaos-manifest), 
+[Azure Chaos Studio](https://azure.microsoft.com/products/chaos-studio), and [Http Fault injector](https://github.com/Azure/azure-sdk-tools/tree/main/tools/http-fault-injector).
```

Comment:
We don't use Chaos Studio, actually. It requires us to install and let them manage parts of our cluster so it's not enabled.